From 21851eaecc814be709cb0c20f732cb858cfe1440 Mon Sep 17 00:00:00 2001
From: Harish Mahendrakar <harish.mahendrakar@ittiam.com>
Date: Tue, 27 Dec 2016 13:54:13 +0530
Subject: Decoder: Fix in MB count in MBAff error handling

Number of MBs to be marked as skip should be even for MBAff
error streams

Bug: 33450635
Change-Id: I6af59632e4b2d2882881fe8c52472164ea088f7a
(cherry picked from commit 43e0aa221067341b0b32bc87410a1ca45d9a6455)
---
 decoder/ih264d_parse_pslice.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/decoder/ih264d_parse_pslice.c b/decoder/ih264d_parse_pslice.c
index 63fa179..9d4e687 100644
--- a/decoder/ih264d_parse_pslice.c
+++ b/decoder/ih264d_parse_pslice.c
@@ -1462,6 +1462,11 @@ WORD32 ih264d_mark_err_slice_skip(dec_struct_t * ps_dec,
         ih264d_err_pic_dispbuf_mgr(ps_dec);
         return 0;
     }
+
+    if(ps_dec->ps_cur_slice->u1_mbaff_frame_flag && (num_mb_skip & 1))
+    {
+        num_mb_skip++;
+    }
     ps_dec->ps_dpb_cmds->u1_long_term_reference_flag = 0;
     if(prev_slice_err == 1)
     {
-- 
cgit v1.1

